### PR TITLE
Project 01: Logic Gates Implementations (Part 2)

### PR DESCRIPTION
**Pull request by wildmaples**
Part 2 out of 2 for [Project 01](https://www.nand2tetris.org/project01). 

Implementations for chips: 

- [x] Not16
- [x] And16
- [x] Mux16
- [x] Or16
- [x] Or8Way
- [x] Mux4Way16
- [x] Mux8Way16
- [x] DMux4Way
- [x] DMux8Way

### CODE REVIEW

**Comment by tomstuart** [comment]
🎉 Both excellent and tedious!

**Comment by wildmaples** [comment]
> Congratulations! This works! 🎉
> 
> There is a simpler solution, if you’re interested. You can feed `sel=sel[0]` into two `Mux16`s — the first one choosing between `a` and `b`, and the second one choosing between `c` and `d`. Then you can feed the outputs of those two into a third `Mux16` with `sel=sel[1]` which will choose either the `a`/`b` option or the `c`/`d` option. Does that make sense? Pretty similar to your solution but only three gates required.

Screen Shot 2021-03-24 at 10 50 07 AM

File: projects/01/DMux8Way.hdl [hunk 1]
```diff
@@ -16,5 +16,7 @@ CHIP DMux8Way {
     OUT a, b, c, d, e, f, g, h;
 
     PARTS:
-    // Put your code here:
-}
\ No newline at end of file
+    DMux(in=in, sel=sel[2], a=sela, b=selb);
+    DMux4Way(in=sela, sel[0]=sel[0], sel[1]=sel[1], a=a, b=b, c=c, d=d);
+    DMux4Way(in=selb, sel[0]=sel[0], sel[1]=sel[1], a=e, b=f, c=g, d=h);
```
Comment (tomstuart) [inline projects/01/DMux8Way.hdl]: FYI there’s a more compact notation for picking out part of a multi-bit bus:

```suggestion
    DMux4Way(in=sela, sel=sel[0..1], a=a, b=b, c=c, d=d);
    DMux4Way(in=selb, sel=sel[0..1], a=e, b=f, c=g, d=h);
```

File: projects/01/Mux8Way16.hdl [hunk 2]
```diff
@@ -18,5 +22,7 @@ CHIP Mux8Way16 {
     OUT out[16];
 
     PARTS:
-    // Put your code here:
-}
\ No newline at end of file
+    Mux4Way16(a=a, b=b, c=c, d=d, sel[0]=sel[0], sel[1]=sel[1], out=abcdsel);
+    Mux4Way16(a=e, b=f, c=g, d=h, sel[0]=sel[0], sel[1]=sel[1], out=efghsel);
```
Comment (tomstuart) [inline projects/01/Mux8Way16.hdl]: Again, this is the same as:

```suggestion
    Mux4Way16(a=a, b=b, c=c, d=d, sel=sel[0..1], out=abcdsel);
    Mux4Way16(a=e, b=f, c=g, d=h, sel=sel[0..1], out=efghsel);
```

**Review by tomstuart — APPROVED** [review]
Fantastic. Well done. 👏🏻

**Comment by tomstuart** [comment]
NICE

**Pull request merged**